extension/app/ts/components/simulationExplaining/identifyTransaction.tsx

```diff
@@ -3,55 +3,82 @@ import { CHAINS, FourByteExplanations, isSupportedChain, MAKE_YOU_RICH_TRANSACTI
 import { SimulatedAndVisualizedTransaction } from '../../utils/visualizer-types.js'
 import { getSwapName, identifySwap } from './SwapTransactions.js'
 
-type TRANSACTION_TYPE = 'MakeYouRichTransaction' | 'NormalTransaction'
+type TRANSACTION_TYPE = 'MakeYouRichTransaction' | 'ArbitaryContractExecution' | 'EtherTransfer' | 'Swap' | 'ContractFallbackMethod'
 
-export function nameTransaction(transaction: SimulatedAndVisualizedTransaction, activeAddress: bigint ) {
-	if (identifyTransaction(transaction, activeAddress) === 'MakeYouRichTransaction') {
-		return 'Simply making you rich'
-	}
-	if (transaction.input.length == 0) return 'Ether Transfer'
-
-	//TODO: add simple token transfer
-
-	const identifiedSwap = identifySwap(transaction)
-	if (identifiedSwap) return getSwapName(identifiedSwap, transaction.chainId)
-
-	const fourByte = get4Byte(transaction.input)
-	if (fourByte === undefined) return 'Contract Fallback Method'
-	const explanation = FourByteExplanations.get(fourByte)
-	return explanation !== undefined ? explanation : 'Contract Execution'
+type IdenttifiedTransaction = {
+	type: TRANSACTION_TYPE,
+	title: string,
+	signingAction: string,
+	simulationAction: string,
+	rejectAction: string,
 }
 
+export function identifyTransaction(transaction: SimulatedAndVisualizedTransaction, activeAddress: bigint): IdenttifiedTransaction {
+	const chainString = transaction.chainId.toString()
 
-export function nameTransactionAction(transaction: SimulatedAndVisualizedTransaction, activeAddress: bigint ) {
-	if (identifyTransaction(transaction, activeAddress) === 'MakeYouRichTransaction') {
-		return 'Rich'
+	if (isSupportedChain(chainString)
+		&& CHAINS[chainString].eth_donator === transaction.from.address
+		&& transaction.to?.address === activeAddress
+		&& transaction.type === MAKE_YOU_RICH_TRANSACTION.type
+		&& transaction.maxFeePerGas === MAKE_YOU_RICH_TRANSACTION.maxFeePerGas
+		&& transaction.maxPriorityFeePerGas === MAKE_YOU_RICH_TRANSACTION.maxPriorityFeePerGas
+		&& transaction.input.toString() === MAKE_YOU_RICH_TRANSACTION.input.toString()
+		&& transaction.value === MAKE_YOU_RICH_TRANSACTION.value
+	) {
+		return {
+			type: 'MakeYouRichTransaction',
+			title: 'Simply making you rich',
+			signingAction: 'Make me rich',
+			simulationAction: 'Simulate richies',
+			rejectAction: 'Reject richies',
+		}
 	}
-	if (transaction.input.length == 0) return 'Ether Transfer'
 
-	//TODO: add simple token transfer
+	if (transaction.input.length == 0) return {
+		type: 'EtherTransfer',
+		title: 'Ether Transfer',
+		signingAction: 'Transfer Ether',
+		simulationAction: 'Simulate Ether Transfer',
+		rejectAction: 'Reject Ether Transfer',
+	}
```

Comment:
This isn't correct I don't think.  There are lots of contracts that have a receive function that takes some action when ETH is received even in the abscence of calldata.

You could *also* check that gasLimit is 21,000, but that is also wrong as you can attach extra gas, and sending ETH to a contract wallet would have a higher gas limit but would also be just a transfer.

At the least, we should track a GitHub issue to revisit this and figure out the best way to identify ETH transfers.  If we simulate first, we could check to see how much gas was *used*, which would notably help identify "complex transaction" vs "simple send".  There is probably some threshold of gas used where "something interesting happened on receipt".

Reply:
ah you are right. We could also check for produced logs

Reply:
A contract wallet may produce logs on ETH received.

Reply:
I mean token logs, if it produces token logs, I won't categorize it as eth send